scheduler/propertyset.go

```diff
@@ -347,3 +362,14 @@ func getProperty(n *structs.Node, property string) (string, bool) {
 
 	return resolveTarget(property, n)
 }
+
+// targetedPropertyValue transforms the property value to combine all implicit
+// target values into a single wildcard placeholder so that we get accurate
+// counts when we compare an explicitly-defined target against multiple implicit
+// targets.
+func (p *propertySet) targetedPropertyValue(propertyValue string) string {
+	if len(p.targetValues) == 0 || slices.Contains(p.targetValues, propertyValue) {
```

Comment:
if this is the key operation you could probably use `set.Set` and make this `O(1)`, though it's also just fine as is

Reply:
The number of values here will typically be really small, but it can't hurt. Done.